src/Application/Config.cpp

```diff
@@ -0,0 +1,84 @@
+//   Copyright 2020 Roddie Kieley
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+#include "Config.h"
+#include "yaml-cpp/node/node.h"
+#include "yaml-cpp/node/parse.h"
+#include "../Logging/loguru.hpp"
+#include <cstring>
+
+// Define the static Singleton pointer
+Config* Config::inst_ = nullptr;
```

Comment:
inst_ -> not sure this style is used anywhere in the code atm, rather the established norm is a somewhat hungarian style. That being said the way the singletons are created in Instance method(s) generally doesn't allocate the instance on the heap to begin with. This should be something like the below which eliminates the requirement to new and delete the instance of the singleton (with associated trade-offs) and as well uses more explicit naming.

 // Singleton
    static MessageConsumer& Instance(_Dependencies* pDependencies)
    {
        static MessageConsumer  theMessageConsumer(pDependencies);
        return theMessageConsumer;
    }

Reply:
Ah yes I found the style guide you seem to be following for member local variable name conventions: http://www1.coe.neu.edu/~ctd/ISY240/CodingGuidelines.htm

But there are other conventions that use a trailing _ for member variables.  I changed this to be consistent with the rest of the code base.